Fix NPE on recipe parsing

garagatyi opened the pull request:
### What does this PR do?
Fix NPE on recipe parsing.

sleshchenko commented on infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java:
```diff
@@ -111,6 +111,7 @@ protected OpenShiftEnvironment doCreate(
         throw new ValidationException("Supporting of deployment configs is not implemented yet.");
       } else if (object instanceof Pod) {
         Pod pod = (Pod) object;
+        checkNotNull(pod.getMetadata(), "Pod metadata must not be null");
```
I think the same should be done for other objects. Also, I see no check that name inside of metadata is not null or empty while it must not be.
Maybe it would be better have somehow util method that will safely retrieve name of `HasMetadata` object?
I mean something like that
```
  public String getName(HasMetadata object) {
    ObjectMeta metadata = object.getMetadata();
    checkNotNull(metadata, String.format("%s metadata must not be null", object.getKind()));
    String name = metadata.getName();
    checkNotNull(metadata, String.format("%s name must not be null", object.getKind()));
    return name;
  }
```
Then we would be able to reuse it for all kinds of K8s/OpenShift objects. WDYT?

skabashnyuk commented:
@sleshchenko @garagatyi Does it make sense to discuss here similar issue in other places or you want to narrow the scope of this pr to ```OpenShiftEnvironmentFactory.java```

sleshchenko commented:
@skabashnyuk Good catch. I think that It's good time to implement other validations rules https://github.com/eclipse/che/issues/7381

garagatyi commented:
@sleshchenko @skabashnyuk I added the same check to k8s recipe parser. Also added the check of name nullness. But the implementation of this is not in my sprint, so I'm not going to expand improvements in this PR. I just wanted to fix NPE we've got in RhChe installation.
